[배민지] sprint2

minji9029 opened the pull request:
# 기본
- [x] "판다마켓" 클릭 시 루트 페이지("/")로 이동합니다. (새로고침)
- [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일합니다.
- [x]  SNS 아이콘들은 클릭시 각각 아래 페이지로 이동합니다. - “[https://www.google.com/”, -](https://www.google.com/%E2%80%9D,%E2%80%A8-) “[https://www.kakaocorp.com/page/”](https://www.kakaocorp.com/page/%E2%80%9D)

# 심화
- [x] 비밀번호 input 요소 오른쪽에 눈모양 아이콘을 추가합니다. (추후 클릭으로 비밀번호를 보거나 가릴 수 있도록 기능을 만들어 갈 예정입니다.)

# 주요 변경사항
- [x] 반응형 추가
- [x] 로그인, 회원가입 페이지 추가

스크린샷
![screencapture-127-0-0-1-5500-FE-Sprint-Mission1-week1-singup-html-2025-04-12-16_51_05](https://github.com/user-attachments/assets/93413830-bd7c-4782-802a-c696acba4493)
![screencapture-127-0-0-1-5500-FE-Sprint-Mission1-week1-login-html-2025-04-12-16_50_38](https://github.com/user-attachments/assets/85bd0cfb-b5e1-4742-b527-06159ac48eb8)
![screencapture-127-0-0-1-5500-FE-Sprint-Mission1-week1-index-html-2025-04-12-16_51_39](https://github.com/user-attachments/assets/b327331e-25bf-423b-9a33-28be5b36616c)


# 질문 
깃을 다루다가 기존에 코드리뷰 주신부분이 날아간듯합니다...
공부도 따로 되어서 원인도 알겠고 앞으로 어떻게 해야할지도 정리가 되었는데
날아간 부분에 대해서 따로 캡쳐둔게있어서 저는 괜찮은데 강사님께서 필요하신 부분인지 걱정이네요 ㅠ

BaeMinJi777 committed:
깃 어렵다...

BaeMinJi777 committed:
Merge branch 'Basic-배민지-sprint1' of github.com:minji9029/16-Sprint-Mission into Basic-배민지-sprint1

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/images/home/feature1-image.png:
투명 배경을 지원할 필요가 없는데 png 파일을 쓰신 이유가 있을까요~?

어떤 경우에 어떤 이미지 포맷을 쓰는게 좋을지에대해 추천드리자면:

- 벡터기반의 장점을 활용할수있거나, path 정보가 간단한 로고, 아이콘등의 파일: svg
- 투명 배경을 지원할 필요가 있는 큰 이미지 파일: png
- 투명 배경을 지원할 필요가 없는 큰 이미지 파일: 최소 2x의 고배율 jpg, jpeg => 고배율 이미지를 써야하므로 png, jpeg보다 압축률이 좋은 webp로 변환해 picture태그와 함께 사용 추천

컨텐츠의 특성 혹은 몇가지 조건에 따라 올바른 파일 포맷을 고르는것도 최적화에 포함됩니다. 
제가 추천드린것말고도 여러가지 참고해보시고 적용해보세요 :)

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/index.html:
```diff
@@ -0,0 +1,185 @@
+      crossorigin
+      href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.9/dist/web/static/pretendard.min.css"
+    />
+    <!-- 스타일시트를 링크해 주세요. styles라는 폴더를 만들어 웹사이트 전역에서 공통으로 사용되는 global 스타일링과 각 페이지 별로 특정적인 스타일링을 위한 파일을 나눠 작성하면 가독성을 개선할 수 있어요. -->
```
요런 주석은 혹시 generator에 의해 생성된걸까요..? 
PR 올리실때는 작업에 대한 설명이 필요한 경우를 제외하고 불필요한 주석을 정리해주세요~!

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/index.html:
```diff
@@ -0,0 +1,185 @@
+    <link rel="stylesheet" href="styles/global.css" />
+    <link rel="stylesheet" href="styles/home.css" />
+  </head>
+
+
+  
```
좋은 포맷팅을 할 수 있도록 노력해봅시다 :)
지금은 습관을 들이기 위해 수동으로 고쳐보고, 나중에 prettier 플러그인을 설치해 일관적으로 포맷팅을 자동화할수있게끔 시도해봐요 :)

prettier가 어떤 도구이고 무엇을 할수있는지 소개해드릴게요!
[참고](https://velog.io/@gangk_99/VS-Code-Prettier-%EC%84%A4%EC%A0%95%ED%95%98%EA%B8%B0)

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/index.html:
```diff
@@ -0,0 +1,185 @@
+        </div>
+      </section>
+
+      <section id="bottomBanner" class="banner">
```
지난번에 코멘트 드렸던것 같은데 bottom이라는 키워드가 빠지게끔 클래스이름 변경해봅시다! :)

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/login.html:
```diff
@@ -0,0 +1,72 @@
+
+  <body class="flex-center"> 
+    <div class="container">
+      <div class="TOP flex-center">
```
TOP도 네이밍 케이스, 네이밍 컨벤션 고려했을때 일관적으로 보일수있게끔 고려해보시면 좋을것같아요 :)

[CSS 네이밍 컨벤션](https://velog.io/@krkorklo58/TIL-CSS-%EB%84%A4%EC%9D%B4%EB%B0%8D-%EC%BB%A8%EB%B2%A4%EC%85%98)
[자주 쓰는 네이밍케이스 정리](https://xionwcfm.tistory.com/459)

두개 참고해보시고, 보다 일관적인 네이밍 규칙을 적용해보시면 좋을 것 같습니다!

addiescode-sj replied:
emailNpassword, emailW도 마찬가지!

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/login.html:
```diff
@@ -0,0 +1,72 @@
+      <form class="flex-center">
+        <div class="emailNpassword">
+          <label for="emailW">이메일</label>
+          <input id="emailW" placeholder="이메일을 입력하세요." type="email">
```
input의 name 속성이 빠져있네요!
name 속성은 폼 데이터 전송 시 필드 이름으로 사용되고,
`<form>`을 통해 데이터를 서버에 전송할 때, 각 input 요소를 식별하는 역할을 합니다.

form 데이터를 서버로 전송하기위해서는 name 속성 사용이 필수적이겠죠?

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/login.html:
```diff
@@ -0,0 +1,72 @@
+        <div class="emailNpassword">
+          <label for="emailW">이메일</label>
+          <input id="emailW" placeholder="이메일을 입력하세요." type="email">
+          <div class="PlusIcon">
```
password와 관련된 wrapper 역할인데 클래스이름은 PlusIcon인게 조금 안맞아보이네요!

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/login.html:
```diff
@@ -0,0 +1,72 @@
+          로그인
+          </button>
+        </a>
+        <div class="loginPlusImg">
```
이 이름도 네이밍케이스가 안맞고, 클래스이름이 좀 모호하게 느껴지는것같아요.
엘리먼트의 역할에 맞게 `social-login` 어떨까요?

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/singup.html:
파일명 오타났네요!

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/styles/global.css:
```diff
@@ -0,0 +1,132 @@
+  font-size: 16px;
+}
+
+#footerMenu {
+  display: flex;
+  gap: 30px;
+  color: #e5e7eb;
+}
+
+#socialMedia {
+  display: flex;
+  gap: 12px;
+}
```
이런 id 선택자를 활용한 CSS 블락들은 페이지에서 쓰여도 한번만 쓰이거나 아예 안쓰일수도있는데, 전역 스타일을 관리하는 파일에 있기보다는 페이지 단위 css 파일에 있는게 적합해요 :)

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/styles/home.css:
```diff
@@ -0,0 +1,75 @@
+  right: 0;
+}
+
+#hero {
+  background-image: url("../images/home/hero-image.png");
+}
+
+#features {
+  padding-bottom: 138px;
+}
+
+#bottomBanner {
+  background-image: url("../images/home/bottom-banner-image.png");
+}
+
+/* 버튼 등의 요소에 고정된 width나 height을 부여하는 것보다 내부 콘텐츠에 padding을 적용해서 요소의 전체 크기를 유동적으로 조정하는 방법이 반응형 웹 디자인과 접근성(accessibility) 측면에서 더 적합합니다. */
+#loginLinkButton {
+  font-size: 16px;
+  font-weight: 600;
+  border-radius: 8px;
+  padding: 11.5px 23px;
+}
```
id를 많이 활용하셨네요! 

id는 특정 요소를 고유하게 식별하기 위해 사용하는 속성으로, 같은 문서내에서 유일한 값을 사용해야합니다. 또한, id 선택자는 클래스나 태그 선택자보다 명시도가 높아 우선 적용되므로 오남용하게되면 스타일을 재정의하거나 수정하기 어렵게 만듭니다. 

이러한 특징 때문에 클래스는 재사용이 가능하고 그룹화하기 쉬운데, id를 사용하면 유지보수성과 확정성의 저하로 이어집니다. 

id를 꼭 필요한 경우에만 사용해볼까요? 참고해두세요! :)

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/styles/login.css:
```diff
@@ -0,0 +1,126 @@
+* {
+  box-sizing: border-box;
+}
+
+:root {
+  --gray-900: #111827;
+  --gray-800: #1f2937;
+  --gray-700: #374151;
+  --gray-600: #4b5563;
+  --gray-500: #6b7280;
+  --gray-400: #9ca3af;
+  --gray-200: #e5e7eb;
+  --gray-100: #f3f4f6;
+  --gray-50: #f9fafb;
+  --blru: #3692ff;
+}
+
+.flex-center {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
요런 전역 스타일 처리는 페이지 단위 css 파일이 아닌 global.css에서 한번만 해주는게 코드 중복을 줄일 수 있는 방법이겠죠?

addiescode-sj commented on FE-Sprint-Mission2/FE-Sprint-Mission1-week1/styles/login.css:
```diff
@@ -0,0 +1,126 @@
+  align-items: center;
+}
+
+body {
+  flex-direction: column;
+  height: 100vh;
+}
```
body 선택자는 신중히 사용하시는게 좋아요.

body에 설정한 스타일은 자식 요소에게 상속되는 경우가 많아요. 
그렇기때문에 필요하지않은 경우에도 body 선택자를 활용해 스타일링하게되면 예상치 못한 오염이 발생할 수 있어요. 또, body 선택자 자체의 명시도가 아주 낮기 때문에 나중에 더 우선순위가 높은 선택자를 사용하게되면 스타일 적용이 안될수있으니, 이걸 유의해주세요.

따라서 지금 같이 **문서 구조 전체가 아닌 특정 레이아웃 영역**만 스타일링 하고 싶다면, body가 아니라 `.page-wrapper`, `.container` 같이 범위가 좁은 클래스명을 쓰는 것을 추천드립니다 :)

addiescode-sj reviewed (approved):
수고하셨습니다!
전체적으로 네이밍과 구조화를 조금 더 신경써보면 훨씬 좋아질것같아요 :)

## 주요 리뷰 포인트
- 네이밍케이스, 네이밍 컨벤션
- 구조화
- 코드 중복 제거
- 선택자 사용 관련 유의사항 (id, body)

BaeMinJi777 committed:
스프린트미션 2 코드리뷰 후 수정

BaeMinJi777 committed:
- 미션2 강의장에 있는 폴더로 작업해야하는줄 알고
폴더에들어있던 예시코드(?)로 직전 PR을 올렸습니다 ㅠ
다시 제가 작성한 코드로 index.html과 css파일로 수정하였습니다.

- 네이밍수정
단편적으로 코드리뷰 주신부분만 수정했었는데
전체적으로 코드를 다시 살펴보며 네이밍 훈련을 하였습니다.

클레스 imgPlusWorde -> 시맨틱태그 figure
emailNpassword -> getImformation
emailW -> getEmail
plusIcon -> wrapper
loginPlusImg -> social-ligin

앞으로도 진행될 미션에서는 네이밍 스타일 통일 방식도 고민하며 작성해보겠습니다.

-불필요한 이미지들을 정리하였고 파일구조가 복잡해보이지 않게 정리하였습니다.

-이미지의 크기를 HTML에 적용해두어 이미지 최적화에 신경써 보았습니다.
